### PR TITLE
Add rust enclave report create and verify wrappers

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
A new trait has been added `mc-sgx-tservice::Report` which provides
functionality to create an `mc-sgx-core-types::Report` and to verify a
`mc-sgx-core-types::Report`.  This functionality operates inside of an
enclave.

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #125** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/125" target="_blank">Graphite</a> 
    * **PR #126** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/126" target="_blank">Graphite</a> 
      * **PR #127** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/127" target="_blank">Graphite</a> 
        * **PR #130** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/130" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/130?utm_source=stack-comment).

File: tservice/src/report.rs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+/// Report operations that can be performed inside of an enclave
+pub trait Report {
+    /// Creates a report for the current enclave.
+    ///
+    /// # Arguments
+    /// * `target_info` - The information for the target enclave which will
+    ///   cryptographically verify the report.  See [`Report::verify_report()`].
+    /// * `report_data` - Report data used to communicate between enclaves
+    fn create_report(
```
Comment (jcape) [inline tservice/src/report.rs:17]: nit: Recommend just `Report::new()` here.

File: tservice/src/report.rs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    /// Verify an enclave report
+    ///
+    /// Ok means that the input report was generated using a [`TargetInfo`]
+    /// that matches the one for the enclave making this call.
+    ///
+    /// # Arguments
+    /// * `report` - The report to verify
+    fn verify_report(report: &mc_sgx_core_types::Report) -> Result<()> {
```
Comment (jcape) [inline tservice/src/report.rs:40]: Recommend `Report::verify()`---the FFIs are `foo_report()`, not `report_foo_report()` :-)

**Review by jcape — APPROVED** [review]
Approved w/ some naming nits.

File: tservice/src/report.rs (hunk 1 above)
Comment (nick-mobilecoin) [reply tservice/src/report.rs]: Converted to new

File: tservice/src/report.rs (hunk 2 above)
Comment (nick-mobilecoin) [reply tservice/src/report.rs]: Renamed to `verify()`

**Pull request merged**